scipy/interpolate/_bsplines.py

```diff
@@ -445,34 +476,89 @@ def integrate(self, a, b, extrapolate=None):
         if extrapolate is None:
             extrapolate = self.extrapolate
 
-        if not extrapolate:
-            # shrink the integration interval, if needed
-            a = max(a, self.t[self.k])
-            b = min(b, self.t[-self.k - 1])
+        # Prepare self.t and self.c.
+        self._ensure_c_contiguous()
 
-            if self.c.ndim == 1:
-                # fast path: use FITPACK's routine (cf _fitpack_impl.splint)
-                t, c, k = self.tck
-                aint, wrk = _dierckx._splint(t, c, k, a, b)
-                return aint
```

Comment:
The idea of this fast path was to avoid computing the antderivative etc in case the answer can be calculated directly. So an easy way would be to adapt the `if not extrapolate` guard to something like `if not extrapolate or (extrapolate !="periodic")`